### PR TITLE
Update `pregenesis.sh` script for testnet4

### PR DESCRIPTION
**Pull request by teddyding**
### Changelist
Update pregenesis.sh script for testnet 4
Generate pregenesis file [here](https://github.com/dydxprotocol/v4-testnets/pull/90)

### Test Plan
N/A

### CODE REVIEW

File: protocol/testing/genesis.sh [hunk 1]
```diff
@@ -22,9 +22,10 @@ NATIVE_TOKEN="adv4tnt" # public testnet token
 DEFAULT_SUBACCOUNT_QUOTE_BALANCE=100000000000000000
 DEFAULT_SUBACCOUNT_QUOTE_BALANCE_FAUCET=900000000000000000
 # Each testnet validator has 1 million whole coins of native token.
-TESTNET_VALIDATOR_NATIVE_TOKEN_BALANCE=1000000$EIGHTEEN_ZEROS # 1e24
+TESTNET_VALIDATOR_NATIVE_TOKEN_BALANCE=1000000$EIGHTEEN_ZEROS # 1e24 or 1 million native tokens.
 # Each testnet validator self-delegates 500k whole coins of native token.
-TESTNET_VALIDATOR_SELF_DELEGATE_AMOUNT=500000$EIGHTEEN_ZEROS # 5e23
+TESTNET_VALIDATOR_SELF_DELEGATE_AMOUNT=500000$EIGHTEEN_ZEROS # 5e23 or 500k native tokens.
+FAUCET_NATIVE_TOKEN_BALANCE=50000000$EIGHTEEN_ZEROS # 5e25 or 50 million native tokens.
```
Comment (teddyding) [inline protocol/testing/genesis.sh:28]: ~60 validators + 3 faucet + 1 rewards vester = 60 million + 150 million + 10 million = 220 million = 22% of total supply.

Note we will need to manually update the bridge module balance after `gentx` process is complete to make it add up to 1 billion total supply

**Review by roy-dydx — APPROVED** [review]

**Review by ttl33 — APPROVED** [review]

File: protocol/testing/genesis.sh [hunk 2]
```diff
@@ -22,16 +22,20 @@ NATIVE_TOKEN="adv4tnt" # public testnet token
 DEFAULT_SUBACCOUNT_QUOTE_BALANCE=100000000000000000
 DEFAULT_SUBACCOUNT_QUOTE_BALANCE_FAUCET=900000000000000000
 # Each testnet validator has 1 million whole coins of native token.
-TESTNET_VALIDATOR_NATIVE_TOKEN_BALANCE=1000000$EIGHTEEN_ZEROS # 1e24
+TESTNET_VALIDATOR_NATIVE_TOKEN_BALANCE=1000000$EIGHTEEN_ZEROS # 1e24 or 1 million native tokens.
 # Each testnet validator self-delegates 500k whole coins of native token.
-TESTNET_VALIDATOR_SELF_DELEGATE_AMOUNT=500000$EIGHTEEN_ZEROS # 5e23
+TESTNET_VALIDATOR_SELF_DELEGATE_AMOUNT=500000$EIGHTEEN_ZEROS # 5e23 or 500k native tokens.
```
Comment (ttl33) [inline protocol/testing/genesis.sh:27]: I think you are missing one zero?

File: protocol/testing/testnet-external/pregenesis.sh [hunk 3]
```diff
@@ -163,19 +161,12 @@ function overwrite_genesis_public_testnet() {
 	dasel put -t string -f "$GENESIS" '.app_state.gov.params.quorum' -v '0.33400' # 33.4%
 	dasel put -t string -f "$GENESIS" '.app_state.gov.params.threshold' -v '0.50000' # 50%
 	dasel put -t string -f "$GENESIS" '.app_state.gov.params.veto_threshold' -v '0.33400' # 33.4%
-
-	# Consensus params
-	dasel put -t string -f "$GENESIS" '.consensus_params.block.max_bytes' -v '22020096'
```
Comment (ttl33) [inline protocol/testing/testnet-external/pregenesis.sh:168]: for my own understanding, why are we removing this?

Comment (ttl33) [reply protocol/testing/testnet-external/pregenesis.sh:168]: nvm, I think this is removed so we can keep the 4mib value that we settled one

**Pull request merged**